spot-client/src/common/remote-control/remoteControlClient.js

```diff
@@ -563,12 +579,12 @@ export class RemoteControlClient extends BaseRemoteControlService {
     _onSpotTvJoinedMuc(spotTvAddress) {
         logger.log('Spot TV joined the MUC', { spotTvAddress });
 
-        // FIXME this doesn't check if the data channels is active. P2P reconnect logic will be added as next step.
-        if (this._p2pSignaling && !this._p2pSignaling.getConnectionForAddress(spotTvAddress)) {
-            // Client initiates the P2P signaling session
-            logger.log('Will try to establish P2P connection with Spot TV');
-            this._p2pSignaling.start(spotTvAddress);
+        if (this._p2pSignaling && !this._spotTvP2PRetry) {
+            this._spotTvP2PRetry = new P2PReconnectTrait(this._p2pSignaling, this.xmppConnection);
```

Comment:
What would you think of making a decorator on top of p2p that handles reconnect instead of having both _p2pSignaling and _spotTvP2PRetry?

Reply:
So this decorator would wrap the P2P signaling and add extra methods?

Reply:
Yeah. p2pSignaling remains the same, just like it is now in the PR, but a new class will take in p2pSignaling (or instantiate it itself) and provide additional methods or functionality on top of p2pSignaling, and only this new class is used by RCS. That way conceptually all the p2p logic should be in this one new class. There will be boring work to do to pass through method calls though. I guess you could also extend the p2pSignaling class to get the same effect. These are just some alternatives I'm pondering right now; I'm not recommending that you actually make changes.

Reply:
Okay, then I got much less excited about the decorator when I heard about `boring work to do to pass through method calls`. Not really sure what benefit would that give other than one less field in RCS.

Reply:
If you would prefer one class then it can be as well moved over to the P2PSignalingClient?

Reply:
The initial reason why I've put it into separate was to avoid piling logics in one file, making it aware of "signaling connection connected/disconnected".